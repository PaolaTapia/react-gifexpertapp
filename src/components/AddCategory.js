import React, {useState} from 'react';
import PropTypes from 'prop-types'; 

 const AddCategory = ({setCategories}) => {

     const [inputValue,setInputValue] = useState (''); 

     const Agregar = (e)=>{
          e.preventDefault(); 
          if(inputValue.trim().length>2){
               setCategories( cats => [ inputValue,...cats]);               
               setInputValue('');
          }
          

     }

     return (
          <form onSubmit={Agregar}>
               <p>Buscar: </p>
               <input
                    type="text"
                    value={inputValue}
                    onChange={(e)=>setInputValue(e.target.value)}                 
               />
          </form>
     )
}
AddCategory.propTypes={

     setCategories: PropTypes.func.isRequired
}

export default AddCategory; 