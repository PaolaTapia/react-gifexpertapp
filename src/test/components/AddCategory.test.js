import React from 'react'; 
import { shallow } from 'enzyme';

import AddCategory from '../../components/AddCategory';

describe('Prueba en AddCategory', ()=> {

     //const setCategories=()=>{}; 
     const setCategories=jest.fn(); //con jest puedo saber como fue llamada, si fue llamada, cuantas veces
     let wrapper; 

     beforeEach(() =>{
          jest.clearAllMocks(); 
          wrapper= shallow(<AddCategory setCategories={setCategories}/>); 

     }); 

     test('debe de mostrarse correctamente', () => {
          expect(wrapper).toMatchSnapshot();           
     }); 

     test('debe de cambiar la caja de texto', () => {
          
          const input=wrapper.find('input'); 
          const value='Hola Mundo'; 

          input.simulate('change',{target:{value //value:value          
          }});           
     });

     test('No debe de postear la informaciÃ³n con submit', () => {
        wrapper.find('form')  .simulate('submit',{preventDefault(){}}); 
        expect(setCategories).not.toHaveBeenCalled(); 
     });   
     
     test('debe de llamar el setCatgories y limpiar la caja de texto', () => {
          //1.simular el inputChange
          const input=wrapper.find('input'); 
          const value='Hola Mundo'; 
          input.simulate('change',{target:{value}});   
          
          //2.simular el submit
          wrapper.find('form')  .simulate('submit',{preventDefault(){}}); 
          
          //3.setCategories se debe haber llamado
          expect(setCategories).toHaveBeenCalled(); 
          expect(setCategories).toHaveBeenCalledTimes(1);           
          expect(setCategories).toHaveBeenCalledWith(expect.any(Function)); 
          
          //4.el valor del input de estar ''          
          expect(input.prop('value')).toBe(''); 
     })
     
         
     
})


