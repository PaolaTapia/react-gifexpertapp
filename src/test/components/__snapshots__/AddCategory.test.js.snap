// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en AddCategory debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Buscar: 
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
